### PR TITLE
use k8s service account to access s3 bucket

### PR DESCRIPTION
**Pull request by ffoysal**
Signed-off-by: Foysal Iqbal <foysal.iqbal.fb@gmail.com>

Once the K8s cluster in AWS infrastructure use can use service account to access aws resources. detail explanation is in https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/ 

In order to use K8s service account according to this need to provide `serviceAccountName` in the pod spec. User does not need to provide `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` if they want to use k8s service account. If they dont want to use service account the default behavior in chart remains as before.

Signed-off-by: Foysal Iqbal <foysal.iqbal.fb@gmail.com>

### CODE REVIEW

File: mender/values.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ global:
     AWS_SECRET_ACCESS_KEY: mysecretkey
     AWS_FORCE_PATH_STYLE: "true"
     AWS_TAG_ARTIFACT: "true"
+    useServiceAccount: "false"
```
Comment (tranchitella) [inline mender/values.yaml:19]: To keep the configuration option consistent with the other ones, I'd rather suggest:
```suggestion
    AWS_SERVICE_ACCOUNT_NAME: ""
```

File: mender/templates/secret-s3-artifacts.yaml [hunk 2]
```diff
@@ -14,5 +14,7 @@ data:
   AWS_URI: {{ .Values.global.s3.AWS_URI | b64enc }}
   AWS_BUCKET: {{ .Values.global.s3.AWS_BUCKET | b64enc }}
   AWS_REGION: {{ .Values.global.s3.AWS_REGION | b64enc }}
+{{- if not .Values.global.s3.useServiceAccount }}
   AWS_AUTH_KEY: {{ .Values.global.s3.AWS_ACCESS_KEY_ID | b64enc }}
   AWS_AUTH_SECRET: {{ .Values.global.s3.AWS_SECRET_ACCESS_KEY | b64enc }}
+{{- end }}
```
Comment (tranchitella) [inline mender/templates/secret-s3-artifacts.yaml:20]:
```suggestion
{{- if not .Values.global.s3.AWS_SERVICE_ACCOUNT_NAME }}
  AWS_AUTH_KEY: {{ .Values.global.s3.AWS_ACCESS_KEY_ID | b64enc }}
  AWS_AUTH_SECRET: {{ .Values.global.s3.AWS_SECRET_ACCESS_KEY | b64enc }}
{{- end }}
```

Pay attention to the missing new lline at the end of the file

File: mender/templates/deployments-deploy.yaml [hunk 3]
```diff
@@ -38,7 +38,9 @@ spec:
       {{- with .Values.deployments.affinity }}
       affinity: {{ tpl (toYaml .) $ | nindent 8 }}
       {{- end }}
-
+{{- if .Values.deployments.serviceAccountName }}
+      serviceAccountName: {{ .Values.deployments.serviceAccountName }}
+{{- end }}
```
Comment (tranchitella) [inline mender/templates/deployments-deploy.yaml:43]:
```suggestion
{{- if .Values.global.s3.AWS_SERVICE_ACCOUNT_NAME }}
      serviceAccountName: {{ .Values.global.s3.AWS_SERVICE_ACCOUNT_NAME }}
{{- end }}
```

**Review by tranchitella — CHANGES_REQUESTED** [review]
See my comments to make the naming of the options consistent.
Please update README.md as well documenting the new options.

**Comment by ffoysal** [comment]
@tranchitella Thank you very much for the feedback. updated accordingly.

**Review by tranchitella — COMMENTED** [review]
@ffoysal thanks, it looks good. Our CI/CD doesn't recognize the Co-authored-by tag, though. I kindly ask you to squash your commits into one, using both the Signed-off-by tag and a Changelog, e.g.

Changelog: Use k8s service account to access s3 bucket

**Comment by ffoysal** [comment]
@tranchitella this branch got dirty, so created a new PR https://github.com/mendersoftware/mender-helm/pull/39
hope that will not have any issue.

**Comment by tranchitella** [comment]
Superseded by https://github.com/mendersoftware/mender-helm/pull/39

**Pull request closed without merging**